### DISCUSSION TITLE
Create Repository from existing folder without syncing present files

### DISCUSSION THREAD

**Discussion by the-tjboss** [community/community#123589] [GitHub Education]
### Hi everyone,

Hi,

I have a fairly large project I'd like to sync on GitHub so I can work from different physical locations.

I would like not to sync any of the initial files (apart from the Git ones), since the project is up to date on all my machines.

I want Git to check out modified file only from this point on, but it automatically checks out all the files from the parent folder (I don't want to use git ignore since I might modify these files later on), and if I "discard" from the desktop app at this point it will delete all the selected files.

Any ideas how I should proceed?

**Comment by Pulkit1822** [comment]
Sure, Here's my guide on creating a repository from an existing folder without syncing present files:

**Creating a Repository from an Existing Folder Without Syncing Present Files**
================================================================================

Hey there! If you have a fairly large project you want to sync on GitHub without uploading all the initial files, follow these steps to achieve that:

### Step 1: Initialize a New Git Repository

Open your terminal and navigate to the folder of your project. Run the following command to initialize a new Git repository without staging any files:

```bash
git init
```

### Step 2: Create a `.gitignore` File

Create a `.gitignore` file in the root of your project folder. Add the paths of the files or folders you don't want to sync to GitHub to this file. This will prevent Git from tracking these files. You can use patterns like `*` to ignore all files or specify specific file paths.

### Step 3: Stage and Commit the `.gitignore` File

Stage the `.gitignore` file for commit using the following command:

```bash
git add .gitignore
```

Commit the changes with a descriptive message:

```bash
git commit -m "Add .gitignore file to exclude certain files"
```

### Step 4: Create a New Repository on GitHub

Go to GitHub and create a new repository without initializing it with a README or any files.

### Step 5: Link the Local Repository to the Remote Repository

Add the URL of the GitHub repository as a remote to your local repository:

```bash
git remote add origin <repository-url>
```

Push the changes to the remote repository:

```bash
git push -u origin master
```

### Step 6: Push Only Modified Files

From now on, Git will only track and push modified files in your project folder. When you make changes and want to push them to GitHub, stage and commit the modified files, then push them to the remote repository.

By following these steps, you can create a new repository on GitHub from your existing folder without syncing the present files. Git will only track and push modified files from this point onward, allowing you to work on your project from different physical locations while controlling which files are synchronized.

**Mark as Correct Answer**
-------------------------

If you found this guide helpful, please mark me as the correct answer. Happy coding!

**Comment by the-tjboss** [comment]
Thanks
